qcodes/utils/validators.py

```diff
@@ -488,32 +488,72 @@ def __repr__(self) -> str:
 
 class Arrays(Validator):
     """
-    Validator for numerical numpy arrays
+    Validator for numerical numpy arrays of numeric types (int, float, complex).
+    By default it validates int and float arrays.
+
+    Min and max validation is not supported for complex numbers.
 
     Args:
         min_value:  Min value allowed, default inf.
         max_value: Max value allowed, default inf.
         shape: The shape of the array, tuple of either ints or Callables taking
             no arguments that return the size along that dim as an int.
+        valid_types: Sequence of types that the validator should support. Should
+            be a subset of the supported types or None. If None all real
+            datatypes will validate.
     """
 
-    validtypes = (int, float, np.integer, np.floating)
+    real_types = (np.integer, np.floating)
+    supported_types = real_types + (np.complexfloating,)
 
     def __init__(self, min_value: numbertypes = -float("inf"),
                  max_value: numbertypes = float("inf"),
-                 shape: TSequence[shape_type] = None) -> None:
+                 shape: TSequence[shape_type] = None,
+                 valid_types: Optional[TSequence[type]] = None) -> None:
 
-        if isinstance(min_value, self.validtypes):
-            self._min_value = min_value
+        if valid_types is not None:
+            for mytype in valid_types:
+
+                is_supported = any(np.issubsctype(mytype, supported_type) for
+                                   supported_type in self.supported_types)
+                if not is_supported:
+                    raise TypeError(f"Array Validator only supports numeric "
+                                    f"types: {mytype} is not supported.")
+
+            self.valid_types = valid_types
         else:
-            raise TypeError('min_value must be a number')
+            self.valid_types = self.real_types
 
-        valuesok = max_value > min_value
+        supports_complex = any(
+            np.issubsctype(my_type, np.complexfloating) for my_type in
+            self.valid_types)
 
-        if isinstance(max_value, self.validtypes) and valuesok:
+        inf_limits = np.isinf(min_value) and np.isinf(max_value)
+
+        if supports_complex and not inf_limits:
+            raise TypeError("min and max_value is not supported for complex "
+                            "types.")
+
+        if any(np.issubsctype(type(min_value), real_type) for real_type in
+               self.real_types):
+            self._min_value = min_value
+        else:
+            raise TypeError('min_value must be a (real) number and the '
+                            'validator must only allow real (non complex) '
+                            'data types')
```

Comment:
The second part of the exception message `and the validator must only allow real (non complex) data types` is not checked in this particulat `if`, but is checked above. Isn't it redundant here?

Also, i guess it might help user if the error message contains smth like `the type of min_value is {type(min_value)}`, m?)

Reply:
It is checked here. Thats why we check against real types.

There are two different cases that needs to be checked. 

* Trying to set min and max (to real numbers) for anything that accepts complex. That is done above.
* Trying to set the min value to a complex number. That is done here.

Reply:
i get this: `* Trying to set the min value to a complex number. That is done here.`. but then why is `and the validator must only allow real (non complex) data types` mentioned in the error message?